src/metabase/pulse.clj

```diff
@@ -80,20 +80,48 @@
     (compare (:row dashcard-1) (:row dashcard-2))
     (compare (:col dashcard-1) (:col dashcard-2))))
 
+(defn- virtual-card-of-type?
+  "Check if dashcard is a virtual with type `ttype`.
+
+  There are currently 3 types of virtual card: text, action, link."
+  [dashcard ttype]
+  (and (map? dashcard)
+       (= ttype (get-in dashcard [:visualization_settings :virtual_card :display]))))
```

Comment:
To make this simpler, I would have made this return the type of the dashcard, so the caller can do an equality or set inclusion. e.g. `(= (virtual-card-type dashcard) "action")`

Reply:
Fair point, I have a follow-up PR that might change this fn, I will do it then.